### DISCUSSION TITLE
Why do we need to create a new CredentialSource when checking credentials.

### DISCUSSION THREAD

**Discussion by sicaboy** [web-auth/webauthn-framework#586] [Q&A] [answered]
Why do we need to create a new CredentialSource when checking credentials.
https://github.com/web-auth/webauthn-framework/blob/670d9e99027d59f204ec879bfbfc21f316684c0f/src/webauthn/src/AuthenticatorAssertionResponseValidator.php#L212
I found every time when user login, it creates a new record in the db.
I tried to remove this line and it works too.

**Comment by sicaboy** [comment, marked as answer]
Solved by passing a non-string $credentialId

**Comment by Spomky** [comment]
You shall pass the object from the DB instead of a string. Passing a string is now deprecated (https://github.com/web-auth/webauthn-framework/blob/670d9e99027d59f204ec879bfbfc21f316684c0f/src/webauthn/src/AuthenticatorAssertionResponseValidator.php#L138)

**Reply by sicaboy** [reply]
Thanks mate
